relay-server/src/metrics_extraction/event.rs

```diff
@@ -12,28 +12,59 @@ use relay_sampling::FieldValueProvider;
 /// extracted. Timestamp and clock drift correction should occur before metrics extraction to ensure
 /// valid timestamps.
 pub fn extract_event_metrics(event: &Event, config: &MetricExtractionConfig) -> Vec<Metric> {
+    extract_metrics_from(event, config)
+}
+
+/// Item from which metrics can be extracted.
+pub trait Extractable {
+    fn category(&self) -> DataCategory;
+    fn timestamp(&self) -> Option<Timestamp>;
+}
+
+impl Extractable for Event {
+    fn category(&self) -> DataCategory {
+        // Obtain the event's data category, but treat default events as error events for the purpose of
+        // metric tagging.
+        match DataCategory::from(self.ty.value().copied().unwrap_or_default()) {
+            DataCategory::Default => DataCategory::Error,
+            category => category,
+        }
+    }
+
+    fn timestamp(&self) -> Option<Timestamp> {
+        self.timestamp.value().copied()
```

Comment:
This is new: Span metrics previously used the _transaction_ end time as timestamp. Now they use the _span_ end time.